staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_filter_test.go

```diff
@@ -0,0 +1,166 @@
+					Verbs:           []string{"*"},
+					NonResourceURLs: []string{"*"},
+				}},
+			}},
+		},
+	}
+	plcObj = &flowcontrol.PriorityLevelConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: plName,
+		},
+		Spec: flowcontrol.PriorityLevelConfigurationSpec{
+			Type: flowcontrol.PriorityLevelEnablementLimited,
+			Limited: &flowcontrol.LimitedPriorityLevelConfiguration{
+				NominalConcurrencyShares: 100,
+				LendablePercent:          &lendable,
+				BorrowingLimitPercent:    &borrowingLimit,
+				LimitResponse: flowcontrol.LimitResponse{
+					Type: flowcontrol.LimitResponseTypeQueue,
+					Queuing: &flowcontrol.QueuingConfiguration{
+						Queues:           10,
+						HandSize:         2,
+						QueueLengthLimit: 10,
+					},
+				},
+			},
+		},
+	}
+	cfgObjs = append(cfgObjs, fsObj, plcObj)
+
+	clientset := clientsetfake.NewSimpleClientset(cfgObjs...)
+	informerFactory := informers.NewSharedInformerFactory(clientset, time.Second)
+	flowcontrolClient := clientset.FlowcontrolV1beta3()
+	clk := eventclock.Real{}
+	controller := newTestableController(TestableConfig{
+		Name:                   "Controller",
+		Clock:                  clk,
+		AsFieldManager:         ConfigConsumerAsFieldManager,
+		FoundToDangling:        func(found bool) bool { return !found },
+		InformerFactory:        informerFactory,
+		FlowcontrolClient:      flowcontrolClient,
+		ServerConcurrencyLimit: 24,
+		RequestWaitLimit:       time.Minute,
+		ReqsGaugeVec:           metrics.PriorityLevelConcurrencyGaugeVec,
+		ExecSeatsGaugeVec:      metrics.PriorityLevelExecutionSeatsGaugeVec,
+		QueueSetFactory:        fqs.NewQueueSetFactory(clk),
+	})
+
+	informerFactory.Start(nil)
+
+	status := informerFactory.WaitForCacheSync(nil)
+	if names := unsynced(status); len(names) > 0 {
+		t.Fatalf("WaitForCacheSync did not successfully complete, resources=%#v", names)
+	}
+
+	go func() {
+		controller.Run(nil)
+	}()
+
+	// ensure that the controller has run its first loop.
+	err := wait.PollImmediate(100*time.Millisecond, 5*time.Second, func() (done bool, err error) {
+		return controller.hasPriorityLevelState(plcObj.Name), nil
+	})
+	if err != nil {
+		t.Errorf("expected the controller to reconcile the priority level configuration object: %s, error: %s", plcObj.Name, err)
+	}
+
+	reqInfo := &request.RequestInfo{
+		IsResourceRequest: false,
+		Path:              "/foobar",
+		Verb:              "GET",
+	}
+	noteFn := func(fs *flowcontrol.FlowSchema, plc *flowcontrol.PriorityLevelConfiguration, fd string) {}
+	workEstr := func() fcrequest.WorkEstimate { return fcrequest.WorkEstimate{InitialSeats: 1} }
+
+	flowUser := testUser{name: "test-user"}
+	rd := RequestDigest{
+		RequestInfo: reqInfo,
+		User:        flowUser,
+	}
+
+	// QueueNoteFn executes twice, once leaving the queue and once exiting the queue, which
+	// should result in ~1s wait time in the latency tracker
+	qnf := fq.QueueNoteFn(func(bool) { time.Sleep(500 * time.Millisecond) })
```

Comment:
would you still get enough signal if you made this 50ms to speed the test up 10x?

Reply:
Fair point -- updated to 50ms (100ms total)

Reply:
Actually, this started to flake for me locally when I reduced to 50ms with +/- 5% margin of error:
```
45s: 1289 runs so far, 2 failures (0.16%)

/tmp/go-stress-20230310T205819-168520030
I0310 20:59:04.906559  417425 apf_controller.go:361] Starting API Priority and Fairness config controller
I0310 20:59:04.907030  417425 apf_controller.go:366] Running API Priority and Fairness config worker
I0310 20:59:04.907101  417425 apf_controller.go:369] Running API Priority and Fairness periodic rebalancing process
--- FAIL: TestQueueWaitTimeLatencyTracker (0.31s)
    apf_filter_test.go:164: unexpected wait time latency, got 108, expected approximately 100ms
FAIL
```

I updated to 250ms as a middle ground and also removed the -5% margin of error since having a latency below the expected should be impossible